snakemake/shell.py

```diff
@@ -58,6 +68,12 @@ def executable(cls, cmd):
                 )
         if os.path.split(cmd)[-1] == "bash":
             cls._process_prefix = "set -euo pipefail; "
+        if ON_WINDOWS and cmd.lower().endswith("bash.exe"):
+            # If bash is selected as the executable on Windows
+            # configure _preocess_prefix and win_quote_cmd for
+            # this to work correctly
+            cls._process_prefix = '"{}" -c'.format(cmd)
+            cls.win_quote_cmd(True)
```

Comment:
Nice surprise :) 

But I think we still need the `bash.exe -c` process prefix because `shell=True` only [works for "cmd.exe" on windows](https://github.com/python/cpython/blob/3.8/Lib/subprocess.py#L1294) and can't be used when setting an external executable.

Reading up on how `CreateProcessW` api works on windows I think we also need to start the command line with the process we are starting, e.i `bash.exe -c "<bash command>"`. I think c programs on windows expect argv[0] to still it self. 

From microsoft's [`CreateProcessW`](https://docs.microsoft.com/en-us/windows/win32/api/processthreadsapi/nf-processthreadsapi-createprocessw) API:

> If both lpApplicationName and lpCommandLine are non-NULL, the null-terminated string pointed to by lpApplicationName specifies the module to execute, and the null-terminated string pointed to by lpCommandLine specifies the command line. The new process can use GetCommandLine to retrieve the entire command line. Console processes written in C can use the argc and argv arguments to parse the command line. Because argv[0] is the module name, C programmers generally repeat the module name as the first token in the command line.

`lpApplication` is the `executable` from python and `lpCommandLine` is the command line (`cmd` in python ))

Reply:
ok!